modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -6781,12 +6780,12 @@ else if (req.changeTopology()) {
                                         "previous [" + previous + "] with timeout " + backwardCheckTimeout);
                                 }
 
-                                liveAddr = checkConnection(new ArrayList<>(nodeAddrs), backwardCheckTimeout);
+                                liveAddr = checkConnection(previous, backwardCheckTimeout);
 
                                 if (log.isInfoEnabled()) {
                                     log.info("Connection check to previous node done: [liveAddr=" + liveAddr
-                                        + ", previousNode=" + U.toShortString(previous) + ", addressesToCheck=" +
-                                        nodeAddrs + ", connectingNodeId=" + nodeId + ']');
+                                        + ", previousNode=" + U.toShortString(previous) + ", connectingNodeId="
```

Comment:
Suggestion: I see warnings in the code if some node doesn't respond. Let's make `U.warn()` if node check failed.

Reply:
Included check on `liveAddr` instance to see whether it's null or not. Based on that condititon, it will shown warning or success.